week-1-ts-ecosystem/w01e03-object-oriented-structure/src/types/product.ts

```diff
@@ -0,0 +1,17 @@
+import {TypeCart} from "../cart/Cart";
+
+
+export interface ProductType<T extends TypeCart> {
+  readonly id: string
+  readonly type: T
+  readonly unitPrice: number
+  readonly currency:CurrencyType
+  name: string
+  quantity:number
+}
+
+export enum CurrencyType {
+  PLN="PLN"
+}
+
+export type updateProductType = Omit<Partial<ProductType<TypeCart>>, 'id'| 'type'| 'unitPrice'| 'currency'>
```

Comment:
Ten typ też powinien być z dużej `UpdateProduct` wystarczy IMO

Reply:
Skoro finalnie zostajesz z `2` wartościami które chcesz updatować, to czytelniej będzie użyć typu w 2 stronę (z użyciem `Pick`):
```typescript
Pick<Partial<ProductType<TypeCart>>, 'name' | 'quantity'>
```